polkadot/xcm/pallet-xcm/src/migration.rs

```diff
@@ -170,6 +170,49 @@ pub mod data {
 		}
 	}
 
+	/// Implementation of `NeedsMigration` for `AuthorizedAliasesMap` data.
+	impl<M: Get<u32>> NeedsMigration for (&VersionedLocation, &BoundedVec<VersionedLocation, M>) {
+		type MigratedData = (VersionedLocation, BoundedVec<VersionedLocation, M>);
+
+		fn needs_migration(&self, _: XcmVersion) -> bool {
+			self.0.identify_version() < XCM_VERSION ||
```

Comment:
Why do we use `XCM_VERSION` here but `new_version` in `try_migrate`? Feels like we should use the same in both

Reply:
Yes, the interface is weird because it allows specifying “new_version” when in practice it should always be the latest version.

Feel free to change and use “XCM_VERSION” (latest) as the target version.

Reply:
Well, the point of `new_version`, is that if we are anybody would need to do a migrations step-by-step, e.g.: 
xcm::vX -> xcm::vX+1
xcm::vX+1 -> xcm::vX-latest
in case of vX -> vX-latest wont be possible.

But if we don't want to be generic and always supports just latest, then I would suggest to change `trait NeedsMigration` and remove version and use always `CurrentXcmVersion`, or if we don't want to change that trait and keep it, then we should use `new_version` for the implementations:
```
fn needs_migration(&self, new_version: XcmVersion)
fn try_migrate(self, new_version: XcmVersion)
```

This `new_version` is propagated from:
```
pub(crate) fn migrate_data_to_xcm_version(
    weight: &mut Weight,
    required_xcm_version: XcmVersion,
)
```

which is used [bellow](https://github.com/paritytech/polkadot-sdk/pull/6336/files#diff-16d7265515a75a97207845364d0320e8f9e3306785a7773930408b2dd7c21f4dR471-R473) by `MigrateToLatestXcmVersion` with `latest`:
```
// migrate other operational data to the latest XCM version in-place
let latest = CurrentXcmVersion::get();
Pallet::<T>::migrate_data_to_xcm_version(&mut weight, latest);
```


Also, we should change visibility to allow calling it outside of the pallet:
```
-pub(crate) fn migrate_data_to_xcm_version(` 
-pub fn migrate_data_to_xcm_version(` 
```